droidlet/tools/crowdsourcing/servermgr/allocate_instances.py

```diff
@@ -146,8 +148,52 @@ def request_instance(instance_num):
     logging.info(f"All {instance_num} are up, ip list: {instance_ips}")
     return instance_ips
 
+
+def register_dashboard_subdomain(cf, zone_id, ip, subdomain):
+    """Registers a unique subdomain for craftassist.io
+    that serves proxied HTTP content using cloudflare.
+    Args:
+    cf -- CloudFlare context with R/W permissions.
+    zone_id -- zone ID used to locate DNS records.
+    ip -- IP of the ECS container that runs dashboard.
+    subdomain -- subdomain contains a unique identifier for this task run, 
+    which is the batch ID concatenated with the run number.
+    """
+    # Check that DNS record does not already exist
+    dns_record_exists = cf.zones.dns_records.get(
+        zone_id, params={"name": "{}.craftassist.io".format(subdomain)}
+    )
+    if dns_record_exists:
+        print("DNS record already exists for {}".format(subdomain))
+        return
+
+    dns_record = {"name": subdomain, "type": "A", "content": ip, "proxied": True}
+    r = cf.zones.dns_records.post(zone_id, data=dns_record)
```

Comment:
we need some error handling here -- use try-catch or check something like status code.